pkg/daemons/control/tunnel.go

```diff
@@ -233,7 +233,7 @@ func (t *TunnelServer) dialBackend(ctx context.Context, addr string) (net.Conn,
 				if n.IsReservedPort(port) {
 					toKubelet = true
 					useTunnel = true
-				} else {
+				} else if t.config.EgressSelectorMode != config.EgressSelectorModeAgent {
 					useTunnel = t.egress[nodeName]
```

Comment:
Thanks for the PR and detailed root-cause! Hopefully it wasn't too hard to get there.

I do wonder if perhaps we should instead have the node remove the egress label when it's not running in pod or cluster mode. Right now it is unconditionally set to true which is probably not correct. That would make the extra check here unnecessary.

https://github.com/k3s-io/k3s/blob/d723775792baaf30eb1bf0455e8455958605c06c/pkg/nodeconfig/nodeconfig.go#L113-L116

Reply:
Sorry for the delay ... I had a ton of work dumped on me shortly after submitting this...

Yes, I think your suggestion makes more sense.  I will adjust this PR.

Reply:
@PaulSD we're going into code freeze for the February releases at the end of the week; do you think you'll have a moment to update the PR before then?